lib/matplotlib/backends/backend_gtk3.py

```diff
@@ -115,6 +115,11 @@ def __init__(self, figure=None):
         style_ctx.add_provider(css, Gtk.STYLE_PROVIDER_PRIORITY_APPLICATION)
         style_ctx.add_class("matplotlib-canvas")
 
+    @classmethod
+    def new_manager(cls, figure, num):
+        # docstring inherited
+        return FigureManagerGTK3(cls(figure), num)
```

Comment:
I find it a bit odd that we code the relation between the canvas and the matching manager into a class method on the canvas.

Two possible suggestions:

1) How about storing this information either in a canvas class attribute, or in global dict? That way, we could have a standalone function `new_manager(canvas_class, figure, num)`, which feels a bit better than having a method on the canvas that creates a manager.

2) Or one goes the alternative route and considers the manager to be an optional part of the canvas.
   ```
   canvas = [make the canvas instance]
   canvas.attach_manager()
   ```
   Possibly there is a better name than attach. But the point here is that you amend an existing canvas instance. The original `new_manager` proposal has the canvas class as entry point, but OTOH you get the manager, that holds a reference to the canvas. In my view this mixes the priority/hierarchy. In the first part, the canvas seems to be the leading object, whereas in the second the manager seems to take over.